### PR TITLE
LPS-191628 Add integration tests to ensure that is possible to send one email with many recipients or many emails with one recipient each

### PR DESCRIPTION
**Pull request by seltong**
Related issue: [LPS-191628](https://liferay.atlassian.net/browse/LPS-191628)

### CODE REVIEW

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 1]
```diff
@@ -63,34 +65,10 @@ public void testSendNotification() throws Exception {
...
-						createNotificationRecipientSetting(
-							"fromName",
-							Collections.singletonMap(
-								LocaleUtil.US, "[%CURRENT_USER_FIRST_NAME%]")),
-						createNotificationRecipientSetting(
-							"to",
-							Collections.singletonMap(
-								LocaleUtil.US,
-								"[%CURRENT_USER_EMAIL_ADDRESS%]"))),
-					ListUtil.toString(getTermNames(), StringPool.BLANK),
```
Comment (seltong) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:90]: I removed the `[%CURRENT_USER_EMAIL_ADDRESS%]` because we have a bug opened related to this. So we can fix the bug and add it here again in the same PR.

What do you think?

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 2]
```diff
@@ -160,14 +226,49 @@ public void testSendNotification() throws Exception {
 			user2.getFirstName(),
 			notificationRecipientSettingsMap.get("fromName"));
 		Assert.assertEquals(
-			user2.getEmailAddress(),
+			expectedSingleRecipient,
+			notificationRecipientSettingsMap.get("singleRecipient"));
+
```
Comment (seltong) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:231]: Email is not sent in the same order every time. The block below is just to cover this scenario.

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 3]
```diff
@@ -47,6 +48,7 @@
 /**
  * @author Feliphe Marinho
  */
+@FeatureFlags("LPS-187854")
```
Comment (murilostodolni) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:51]: Could you separate the add FF in another commit, please?

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java (hunk 1 above)
Comment (murilostodolni) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:90]: For me ok, but we need to put this comment in the related ticket to register that is necessary come back the `[%CURRENT_USER_EMAIL_ADDRESS%]` to the test.

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java (hunk 2 above)
Comment (murilostodolni) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:231]: So, in this case, the Assert:
```java
Assert.assertEquals(
			Arrays.toString(actualEmailAddresses),
			expectedEmailAddresses.length, actualEmailAddresses.length);
```
is it not enough?

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 4]
```diff
@@ -127,18 +200,11 @@ public void testSendNotification() throws Exception {
...
-				NotificationConstants.TYPE_EMAIL,
-				NotificationQueueEntryConstants.STATUS_SENT);
-
-		Assert.assertEquals(
-			notificationQueueEntries.toString(), 1,
-			notificationQueueEntries.size());
-
-		NotificationQueueEntry notificationQueueEntry =
-			notificationQueueEntries.get(0);
+	private void _assertNotificationRecipientSettings(
```
Comment (murilostodolni) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:205]: This method is used once and inside another private method. I believe we can add this logic in `_assetNotificationQueueEntry`. What do you think?

**Review by murilostodolni — COMMENTED** [review]
Hey @marcelabc 

As @seltong is in vacation. Could you please review my comments? I will make the changes if necessary.

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java (hunk 4 above)
Comment (marcelabc) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:205]: Agree.

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java (hunk 1 above)
Comment (marcelabc) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:90]: Agree.

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java (hunk 3 above)
Comment (marcelabc) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:51]: I leave it up to you to decide this :+1:

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java (hunk 2 above)
Comment (marcelabc) [reply modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:231]: I would let both Asserts in the test because one is testing the property (true/false) and the other is checking the size of the recipients. What do you think?

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java [hunk 5]
```diff
@@ -160,14 +226,49 @@ public void testSendNotification() throws Exception {
...
+		String[] expectedEmailAddresses = StringUtil.split(expectedToRecipient);
 
 		Assert.assertEquals(
-			String.valueOf(new InternetAddress(user2.getEmailAddress())),
-			mailMessage.getFirstHeaderValue("To"));
+			Arrays.toString(actualEmailAddresses),
+			expectedEmailAddresses.length, actualEmailAddresses.length);
+
+		for (String expectedEmailAddress : expectedEmailAddresses) {
+			boolean contains = false;
```
Comment (marcelabc) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java:244]: I would rename for `containsEmailAddress`. What do you think?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by marcelabc** [comment]
Hey @murilostodolni,
Thank you so much for helping with this issue. Let me know if you need help :+1:

**Comment by murilostodolni** [comment]
resent to: https://github.com/liferay-objects/liferay-portal/pull/2442
